vars/unitTest.groovy

```diff
@@ -84,12 +87,14 @@ def call(Map config = [:]) {
     stashes.add("${target_compiler}-install")
     stashes.add("${target_compiler}-build-vars")
   }
+  
+  def valgrind = config.get('valgrind', '')
 
   Map params = [:]
   params['stashes'] = stashes
   params['script'] = "SSH_KEY_ARGS=${env.SSH_KEY_ARGS} " +
                      "NODELIST=${nodelist} " +
-                     test_script
+                     test_script + " " + valgrind
```

Comment:
Pretty sure that is not going to work.  This is appending to the end of a passed script file, not a command.

There may or may not be a newline at the end of the test_script.  No way to predict what a future change might do even if it works now.